### ISSUE TITLE
App Center SDK as a Swift Package for Xcode 11

### ISSUE THREAD

**Issue by shpakovski** [microsoft/appcenter#1295]
WWDC’19 is a fantastic event. Among other things, Apple is adding **native** support for Swift Packages to Xcode 11. This makes Swift Package Manager an officially supported package manager for third-party libraries on all platforms, including iOS. What’s your plan regarding CocoaPods, Carthage and the future of SDK? Thanks.

**Comment by huseyinbagana** [comment]
Is there any update in this feature ?

**Comment by JoshHrach** [comment]
We would also like to see Swift Package Manager support for App Center.

**Comment by shpakovski** [comment]
Hey Microsoft 👋 Any (good) news? Is this even in plans? Thanks.

**Comment by huseyinbagana** [comment]
Is there any update?

**Comment by ahdbilal** [comment]
This is under active consideration. We will share more details once we get approval and the team moves into implementation phase.

**Comment by kaunteya** [comment]
Any update on this?

AppCenter is the only pod in my Podfile now.

**Comment by ahdbilal** [comment]
Thanks for the vote. We are seriously looking into it. I will share an update soon. Thanks

**Comment by ahdbilal** [comment]
The team has picked this up and are now evaluating it. There are few limitations and we might have to release this in phases. I will share an update once the evaluation is complete.

**Comment by JoshHrach** [comment]
@ahdbilal Thanks for the update. I'm assuming a lot more will be possible with Swift 5.3 and the improvements coming to Swift Package Manager.

https://github.com/apple/swift-evolution/blob/master/proposals/0271-package-manager-resources.md
https://github.com/apple/swift-evolution/blob/master/proposals/0272-swiftpm-binary-dependencies.md
https://github.com/apple/swift-evolution/blob/master/proposals/0273-swiftpm-conditional-target-dependencies.md

**Comment by MatkovIvan** [comment]
Hi there,
The feature has been [merged](https://github.com/microsoft/appcenter-sdk-apple/pull/2039) to `develop` branch, so you can already try it. This will be included in the next SDK release. The feedback is always welcome.

Thanks

**Comment by kaunteya** [comment]
@MatkovIvan Is this still in beta?

**Comment by MatkovIvan** [comment]
@kaunteya there are no releases with it yet, so you need to specify branch or commit hash for now. You can treat this as a beta, yes.

**Comment by cliftonlabrum** [comment]
I just tried this by specifying the `master` branch in **Xcode 11.4.1**:

<img width="721" alt="ac1" src="https://user-images.githubusercontent.com/2513753/81841593-8ceaf080-94ff-11ea-8d1c-81266aebbc04.png">

...but I'm getting an error:

<img width="723" alt="ac2" src="https://user-images.githubusercontent.com/2513753/81841690-abe98280-94ff-11ea-8a91-242eb8a5387d.png">

Do I have to specify a branch other than `master`?

**Comment by kaunteya** [comment]
> Hi there,
> The feature has been [merged](https://github.com/microsoft/appcenter-sdk-apple/pull/2039) to `develop` branch, so you can already try it. This will be included in the next SDK release. The feedback is always welcome.
> 
> Thanks

Try develop branch

**Comment by cliftonlabrum** [comment]
Thanks for your quick reply, @kaunteya. I just tried that and get a different error now. :) Do I need to specify a particular commit?

I have:

1. Removed the Carthage version of App Center
2. Deleted derived data
3. Restarted Xcode

Any ideas?

<img width="485" alt="ac-develop" src="https://user-images.githubusercontent.com/2513753/81848708-61b9ce80-950a-11ea-9882-f7f5edabd48d.png">
<img width="655" alt="ac-error" src="https://user-images.githubusercontent.com/2513753/81848748-70a08100-950a-11ea-97ff-b37ca69b1460.png">

**Comment by cliftonlabrum** [comment]
I figured it out. You have to use the Apple SDK repo path.

**Repository:** `https://github.com/microsoft/appcenter-sdk-apple`
**Branch:** : `develop`

**Comment by tonyarnold** [comment]
I can see this has been released as part of v3.2.0, but I can't find any documentation for the feature. I'm seeing the following error when I attempt to build a macOS app target that integrates both dependencies:

<img width="637" alt="Screen Shot 2020-05-25 at 14 28 11" src="https://user-images.githubusercontent.com/4802/82778180-f79f1300-9e93-11ea-8d2b-1c86ed70d617.png">

**Comment by MatkovIvan** [comment]
Hi there,

This has been released in [3.2.0](https://github.com/microsoft/appcenter-sdk-apple/releases/tag/3.2.0).

@tonyarnold documentation changes is already merged and will be published soon.
Regarding "cyclic" dependencies - core module doesn't have this dependency for sure. I saw the similar issues in Xcode because of "Find Implicit Dependencies" checkbox (Edit scheme -> Build), please try to disable it. If the problem persists - please open a new issue in the apple repo, thanks.
